leftwm-core/src/handlers/command_handler/scratchpad_handler.rs

```diff
@@ -0,0 +1,1211 @@
+
+            Some(manager.command_handler(&Command::SendWindowToTag {
+                window: Some(window_handle),
+                tag: destination_tag,
+            }))
+        }
+        ReleaseScratchPadOption::ScrathpadName(scratchpad_name) => {
+            // Remove and get value from active_scratchpad
+            let window_pid = manager
+                .state
+                .active_scratchpads
+                .get_mut(&scratchpad_name)
+                .and_then(|pids| {
+                    next_valid_scratchpad_pid(pids, &manager.state.windows, Direction::Forward)
+                })?;
+            manager // We found already a working pid, discard from scratchpad
+                .state
+                .active_scratchpads
+                .get_mut(&scratchpad_name)?
+                .pop_front();
+
+            let window_handle = manager
+                .state
+                .windows
+                .iter()
+                .find(|w| w.pid == Some(window_pid))
+                .map(|w| w.handle);
+
+            log::debug!(
+                "Releasing scratchpad {} to tag {}",
+                scratchpad_name,
+                destination_tag
+            );
+
+            Some(manager.command_handler(&Command::SendWindowToTag {
+                window: window_handle,
+                tag: destination_tag,
+            }))
+        }
+        ReleaseScratchPadOption::None => unreachable!(), // Should not be possible
+    }
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Direction {
+    Forward,
+    Backward,
+}
+
+/// Cycles the currently visible scratchpad window given the scratchpads name. Only visible
+/// scratchpads will be handled, otherwise ignored
+pub fn cycle_scratchpad_window<C: Config, SERVER: DisplayServer>(
+    manager: &mut Manager<C, SERVER>,
+    scratchpad_name: &str,
+    direction: Direction,
+) -> Option<bool> {
+    // Prevent cycles when scratchpad is not visible
+    if !is_scratchpad_visible(manager, scratchpad_name) {
+        return Some(false);
+    }
+
+    let scratchpad = manager.state.active_scratchpads.get_mut(scratchpad_name)?;
+    // Get a handle to the currently visible window, so we can hide it later
+    let visible_window_handle = manager
+        .state
+        .windows
+        .iter()
+        .find(|w| w.pid.as_ref() == scratchpad.front()) // scratchpad.front() ok because checked in is_scratchpad_visible
+        .map(|w| w.handle);
+
+    // Reorder the scratchpads
+    // Clean scratchpad and exit if no next exists
+    next_valid_scratchpad_pid(scratchpad, &manager.state.windows, direction)?;
+    match direction {
+        Direction::Forward => {
+            // Perform cycle
+            let front = scratchpad.pop_front()?;
+            scratchpad.push_back(front);
+        }
+        Direction::Backward => {
+            // Perform cycle
+            let back = scratchpad.pop_back()?;
+            scratchpad.push_front(back);
+        }
+    };
```

Comment:
I think we can make use of [`rotate_left`](https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.rotate_left) and [`rotate_right`](https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.rotate_right):
```rust
    match direction {
        Direction::Forward => scratchpad.rotate_left(1),
        Direction::Backward => scratchpad.rotate_right(1),
    };
```

Reply:
Didn't know this existed, will adjust :smile: